### PR TITLE
SQIL and PC performance check fixes

### LINKED ISSUE #791: tests/algorithms/test_sqil.py::test_sqil_performance_continuous[DDPG] failure
## Bug description
```
>       assert reward_improvement.is_significant_reward_improvement(
            rewards_before,  # type:ignore[arg-type]
            rewards_after,  # type:ignore[arg-type]
        )
E       assert False
E        +  where False = <function is_significant_reward_improvement at 0x135cafdc0>([-1379.139569, -1592.006733, -1771.549952, -1345.075819, -1701.436911, -1223.385152, ...], [-1331.957287, -1518.792578, -1274.336042, -1294.565801, -1089.05092, -1652.311479, ...])
E        +    where <function is_significant_reward_improvement at 0x135cafdc0> = reward_improvement.is_significant_reward_improvement
```

Relevant CircleCI link: https://app.circleci.com/pipelines/github/HumanCompatibleAI/imitation/3903/workflows/84d03683-66fc-416e-865a-27bb86d78349/jobs/15818

## Steps to reproduce
Directly run at master head.
`pytest tests/algorithms/test_sqil.py::test_sqil_performance_continuous[DDPG]`

### LINKED ISSUE #807: Fix flaky SQIL test
## Bug description
SQIL test `test_sqil_performance_continuous[TD3]` is failing on Windows, [example error](https://app.circleci.com/pipelines/github/HumanCompatibleAI/imitation/3984/workflows/2e751eaa-5359-49a3-a3c0-87c3981362de/jobs/16360/parallel-runs/0/steps/0-113):
```
================================== FAILURES ===================================
____________________ test_sqil_performance_continuous[TD3] ____________________
[gw5] win32 -- Python 3.8.10 C:\Users\circleci.PACKER-64C83029\project\venv\Scripts\python.exe

rng = Generator(PCG64) at 0x1893B35D660
pytestconfig = <_pytest.config.Config object at 0x000001896826F280>
pendulum_single_venv = <stable_baselines3.common.vec_env.dummy_vec_env.DummyVecEnv object at 0x000001893B63CBE0>
rl_algo_class = <class 'stable_baselines3.td3.td3.TD3'>

    @pytest.mark.parametrize("rl_algo_class", RL_ALGOS_CONT_ACTIONS)
    def test_sqil_performance_continuous(
        rng: np.random.Generator,
        pytestconfig: pytest.Config,
        pendulum_single_venv: vec_env.VecEnv,
        rl_algo_class: Type[off_policy_algorithm.OffPolicyAlgorithm],
    ):
        rl_kwargs = dict(
            learning_starts=500,
            learning_rate=0.001,
            gamma=0.95,
            seed=42,
        )
        if rl_algo_class == ddpg.DDPG:
            rl_kwargs["gamma"] = 0.99
            rl_kwargs["learning_starts"] = 100
>       _test_sqil_performance(
            rng,
            pytestconfig,
            pendulum_single_venv,
            "Pendulum-v1",
            rl_algo_class=rl_algo_class,
            rl_kwargs=rl_kwargs,
        )

tests\algorithms\test_sqil.py:258: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

rng = Generator(PCG64) at 0x1893B35D660
pytestconfig = <_pytest.config.Config object at 0x000001896826F280>
venv = <stable_baselines3.common.vec_env.dummy_vec_env.DummyVecEnv object at 0x000001893B63CBE0>
env_name = 'Pendulum-v1'
rl_algo_class = <class 'stable_baselines3.td3.td3.TD3'>
rl_kwargs = {'gamma': 0.95, 'learning_rate': 0.001, 'learning_starts': 500, 'seed': 42}

    def _test_sqil_performance(
        rng: np.random.Generator,
        pytestconfig: pytest.Config,
        venv: vec_env.VecEnv,
        env_name: str,
        rl_algo_class: Type[off_policy_algorithm.OffPolicyAlgorithm],
        rl_kwargs: Optional[Dict[str, Any]] = None,
    ):
        SEED = 42
        demonstrations = get_demos(env_name, rng, pytestconfig, "transitions")
        model = sqil.SQIL(
            venv=venv,
            demonstrations=demonstrations,
            policy="MlpPolicy",
            rl_algo_class=rl_algo_class,
            rl_kwargs=rl_kwargs,
        )
    
        venv.seed(SEED)
        rewards_before, _ = evaluate_policy(
            model.policy,
            venv,
            100,
            return_episode_rewards=True,
        )
    
        model.train(total_timesteps=10_000)
    
        venv.seed(SEED)
        rewards_after, _ = evaluate_policy(
            model.policy,
            venv,
            100,
            return_episode_rewards=True,
        )
    
>       assert reward_improvement.is_significant_reward_improvement(
            rewards_before,  # type:ignore[arg-type]
            rewards_after,  # type:ignore[arg-type]
        )
E       assert False
E        +  where False = <function is_significant_reward_improvement at 0x0000018916DB8B80>([-1379.139568, -1592.006733, -1771.549952, -1345.075819, -1701.436911, -1223.385152, ...], [-1634.74416, -1042.73655, -1502.465151, -1659.920304, -1518.538321, -950.251335, ...])
E        +    where <function is_significant_reward_improvement at 0x0000018916DB8B80> = reward_improvement.is_significant_reward_improvement
```

I suspect it's seed dependent and Windows PRNG differs from Mac/Linux, so it crops up on that platform but not others.

Should correct test flakiness perhaps by hyperparameter tuning to improve stability, or if we can't make it not flaky better just to delete the test.

### PR DESCRIPTION
**Pull request by ernestum**
## Description

- Speeds up SQIL tests by decreasing the number of training steps 
- Skips SQIL performance tests because they are flaky
- Adds some more seeds to SQIL tests
- Fixes failing preference comparison performance test by increasing number of training steps and the number of samples to compare trained and novice agent.

Fixes #807 
Fixes #791 

## Testing

Changes are covered by the existing tests.

### CODE REVIEW

**Comment by ernestum** [comment]
Hi @AdamGleave would you review and merge? The coverage went down since I disabled the SQIL performance tests for now.

**Review by AdamGleave — APPROVED** [review]
LGTM. Sorry for the slow review. If I'm blocking something feel free to either ping me or get someone else to review (can start asking @dan-pandori for example).
